dali/nvimgcodec/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2023-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Generate necessary stub for nvimgcodec if needed
+
+DETERMINE_GCC_SYSTEM_INCLUDE_DIRS("c++" "${CMAKE_CXX_COMPILER}" "${CMAKE_CXX_FLAGS}" INFERRED_COMPILER_INCLUDE)
+
+# transform a list of paths into a list of include directives
+set(DEFAULT_COMPILER_INCLUDE)
+foreach(incl_dir ${INFERRED_COMPILER_INCLUDE})
+  set(DEFAULT_COMPILER_INCLUDE "${DEFAULT_COMPILER_INCLUDE} -I${incl_dir}")
+endforeach(incl_dir)
+separate_arguments(DEFAULT_COMPILER_INCLUDE UNIX_COMMAND  "${DEFAULT_COMPILER_INCLUDE}")
```

Comment:
Do we need this kind of magic stuff? How do we do it with other projects?

Reply:
We need to extract paths to headers when we cross-compile. A stub generator that uses clang under the hood will use host paths by default.